native/src/main/java/io/ballerina/wso2/apim/catalog/ServiceCatalog.java

```diff
@@ -0,0 +1,265 @@
+import io.ballerina.runtime.api.creators.TypeCreator;
+import io.ballerina.runtime.api.creators.ValueCreator;
+import io.ballerina.runtime.api.types.ArrayType;
+import io.ballerina.runtime.api.types.ObjectType;
+import io.ballerina.runtime.api.types.RecordType;
+import io.ballerina.runtime.api.types.Type;
+import io.ballerina.runtime.api.utils.StringUtils;
+import io.ballerina.runtime.api.utils.TypeUtils;
+import io.ballerina.runtime.api.values.BArray;
+import io.ballerina.runtime.api.values.BMap;
+import io.ballerina.runtime.api.values.BObject;
+import io.ballerina.runtime.api.values.BString;
+import io.ballerina.wso2.apim.catalog.utils.Constants;
+import io.swagger.v3.core.util.Yaml;
+import io.swagger.v3.oas.models.OpenAPI;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+
+import static io.ballerina.wso2.apim.catalog.utils.Constants.HTTP_ORG_NAME;
+import static io.ballerina.wso2.apim.catalog.utils.Utils.createMd5Hash;
+import static io.ballerina.wso2.apim.catalog.utils.Utils.generateBasePath;
+import static io.ballerina.wso2.apim.catalog.utils.Utils.getDefinitionType;
+import static io.ballerina.wso2.apim.catalog.utils.Utils.getHostname;
+import static io.ballerina.wso2.apim.catalog.utils.Utils.getHttpAnnotation;
+import static io.ballerina.wso2.apim.catalog.utils.Utils.getModuleAnnotation;
+import static io.ballerina.wso2.apim.catalog.utils.Utils.getOpenApiDefinition;
+import static io.ballerina.wso2.apim.catalog.utils.Utils.getSecurityType;
+import static io.ballerina.wso2.apim.catalog.utils.Constants.COLON;
+import static io.ballerina.wso2.apim.catalog.utils.Constants.CONFIG;
+import static io.ballerina.wso2.apim.catalog.utils.Constants.DEFAULT_STRING;
+import static io.ballerina.wso2.apim.catalog.utils.Constants.DEFINITION_FILE_CONTENT;
+import static io.ballerina.wso2.apim.catalog.utils.Constants.DEFINITION_TYPE;
+import static io.ballerina.wso2.apim.catalog.utils.Constants.DEFINITION_URL;
+import static io.ballerina.wso2.apim.catalog.utils.Constants.DESCRIPTION;
+import static io.ballerina.wso2.apim.catalog.utils.Constants.LOCALHOST;
+import static io.ballerina.wso2.apim.catalog.utils.Constants.MD5;
+import static io.ballerina.wso2.apim.catalog.utils.Constants.MUTUAL_SSL;
+import static io.ballerina.wso2.apim.catalog.utils.Constants.MUTUAL_SSL_ENABLED;
+import static io.ballerina.wso2.apim.catalog.utils.Constants.NAME;
+import static io.ballerina.wso2.apim.catalog.utils.Constants.NONE;
+import static io.ballerina.wso2.apim.catalog.utils.Constants.PORT;
+import static io.ballerina.wso2.apim.catalog.utils.Constants.SECURE_SOCKET;
+import static io.ballerina.wso2.apim.catalog.utils.Constants.SECURITY_TYPE;
+import static io.ballerina.wso2.apim.catalog.utils.Constants.SERVICE_KEY;
+import static io.ballerina.wso2.apim.catalog.utils.Constants.SERVICE_URL;
+import static io.ballerina.wso2.apim.catalog.utils.Constants.VERSION;
+
+public class ServiceCatalog {
+
+    public static BArray getArtifacts(Environment env) {
```

Comment:
@HindujaB
we are repeating ourself with this https://github.com/ballerina-platform/module-ballerinax-wso2.controlplane/blob/2d1aff4bb94fa19cef8b6d671a124678a551dedb/native/src/main/java/io/ballerina/lib/wso2/controlplane/Artifacts.java#L64

Looks like we need a ballerina API for this.

Reply:
Had a chat with @warunalakshitha and seems like this is the better way.(using lang dependency)